### PR TITLE
Cloud provider cleanup #6464

### PR DESCRIPTION
**Pull request by mikeorzel2**
Initial commit to cover issue #6464. This PR doesnt change add any features or functionality and simply reorders the code to be more readable. The next PR will look to standardise naming across the cloud provider interfaces.

### CODE REVIEW

**Comment by mikeorzel2** [comment]
I signed it! (Note that this is under a corporate CLA under the company name Emerging Stack Pty Ltd

Thanks

Mike

File: pkg/cloudprovider/aws/aws.go [hunk 1]
```diff
@@ -34,87 +34,62 @@ import (
 	"github.com/golang/glog"
 )
 
-// Abstraction over EC2, to allow mocking/other implementations
-type EC2 interface {
-	// Query EC2 for instances matching the filter
-	Instances(instIds []string, filter *ec2InstanceFilter) (resp *ec2.InstancesResp, err error)
-}
+/****************************************************************************************************
```
Comment (gmarek) [inline pkg/cloudprovider/aws/aws.go]: We generally don't use /* */ comments in Kubernetes code, as it's often a pain during debugging. Please change to // here and everywhere below.

**Comment by gmarek** [comment]
LGTM, except the comment format. I think I liked the "original" (read: from before organic growth, not the current one) idea of having types defined at the beginning of the file, but your ordering also makes a lot of sense. 

It's sooo much better when everything is in the same order that I even don't know how to express this. Thanks for doing it.

**Comment by mikeorzel2** [comment]
Hi,

No problem, happy to help. Will update to the suggested comment style tommorrow evening AU time and do another push.

Hopefully this is the first step to cleaning it up further but having it in a consistent format makes it much clearer and easier to compare approaches between providers I found.

Thanks

Mike

**Comment by mikeorzel2** [comment]
Hi,

The comment styles are now as requested. As per previous push, no code changes.

Thanks

Mike

File: pkg/cloudprovider/vagrant/vagrant.go [hunk 2]
```diff
@@ -31,6 +31,42 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/cloudprovider"
 )
 
+/****************************************************************************************************
```
Comment (gmarek) [inline pkg/cloudprovider/vagrant/vagrant.go]: Comments in this file are still in the old format.

**Comment by gmarek** [comment]
Also please squash commits when you fix vagrant.go

**Comment by mikeorzel2** [comment]
Hi,

Commits now squashed, can see that shippable build is failing but cant see details.

Thanks

Mike

**Comment by piosz** [comment]
Shippable looks ok. Will merge once I verify CLA. Could you please send me an email from Emerging Stack Pty Ltd domain? My ldap is pszczesniak and the domain is google.com.

**Comment by mikeorzel2** [comment]
Email sent as requested.

**Comment by erictune** [comment]
@piosz were you able to determine CLA status?

**Comment by piosz** [comment]
I'm in progress.

File: pkg/cloudprovider/openstack/openstack.go [hunk 3]
```diff
@@ -208,123 +218,223 @@ func (os *OpenStack) Instances() (cloudprovider.Instances, bool) {
...
+
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Anything left in addrs is a new member that needs to be added
+	for addr := range addrs {
+		_, err := members.Create(lb.network, members.CreateOpts{
```
Comment (satnam6502) [inline pkg/cloudprovider/openstack/openstack.go]: Combine with following `if` i.e.
```
if _, err := .... ; err != nil {
  return err
}
```
Likewise elsewhere.

File: pkg/cloudprovider/gce/gce.go [hunk 4]
```diff
@@ -42,75 +42,19 @@ import (
...
+// - New* Cloud function
+//===================================================================================================
 
-type Config struct {
-	Global struct {
-		TokenURL  string `gcfg:"token-url"`
-		ProjectID string `gcfg:"project-id"`
-	}
-}
+const EXTERNAL_IP_METADATA_URL = "http://169.254.169.254/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip"
```
Comment (satnam6502) [inline pkg/cloudprovider/gce/gce.go]: What is this hard coded IP?

**Comment by piosz** [comment]
CLA status ok.

**Comment by piosz** [comment]
@mikeorzel2 sorry for the delay. I needed to verify your CLA. Could you please rebase to resolve conflicts?

**Comment by piosz** [comment]
ping @mikeorzel2

**Comment by piosz** [comment]
ping @mikeorzel2

**Comment by piosz** [comment]
Sorry @mikeorzel2, I have to close this PR. If you want to still merge this feel free to reopen and ping me.

**Pull request closed without merging**